tests/system/src/test/java/test/javafx/embed/swing/JDK8220484Test.java

```diff
@@ -0,0 +1,155 @@
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.embed.swing;
+
+import javafx.application.Platform;
+import javafx.embed.swing.JFXPanel;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import junit.framework.AssertionFailedError;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.lang.reflect.Field;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static test.util.Util.TIMEOUT;
+
+public class JDK8220484Test {
+    static CountDownLatch launchLatch;
+
+    private static MyApp myApp;
+
+    private static Timer t;
+    static int cnt;
+
+    @BeforeClass
+    public static void setupOnce() {
+        System.setProperty("sun.java2d.uiScale.enabled", "true");
+        System.setProperty("sun.java2d.uiScale", "125%");
+        System.setProperty("glass.win.uiScale", "1.25");
+        System.setProperty("glass.gtk.uiScale", "1.25");
+        launchLatch = new CountDownLatch(1);
+
+        // Start the Application
+        SwingUtilities.invokeLater(() -> {
+            myApp = new MyApp();
+        });
+
+        try {
+            if (!launchLatch.await(5 * TIMEOUT, TimeUnit.MILLISECONDS)) {
+                throw new AssertionFailedError("Timeout waiting for Application to launch (" + (5 * TIMEOUT) + " seconds)");
+            }
+        } catch (InterruptedException ex) {
+            AssertionFailedError err = new AssertionFailedError("Unexpected exception");
+            err.initCause(ex);
+            throw err;
+        }
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        Platform.exit();
+    }
+
+    @Test
+    public void testScale() throws Exception {
+        // Get the Swing-side BackBuffer
+        Field fpixelsIm = JFXPanel.class.getDeclaredField("pixelsIm");
+        fpixelsIm.setAccessible(true);
```

Comment:
This isn't the pattern we use to access internal fields of a JavaFX class, and won't work.

We typically use the "shim" pattern for such white-box testing. Can you look into adding shims to the `javafx.swing` module? Many of the other modules already have shims, so you can use that as a pattern. It will require adding a package-scope `test_getPixelsIm` method to `JFXPanel`.

Alternatively, you can use AWT `Robot` to read the JFXPanel pixels.

Reply:
I've added a shim. I would like to check the backing image directly.